### PR TITLE
feat: SDA-2393 Adding/removing registry keys

### PR DESCRIPTION
**Pull request by mattias-symphony**
## Description
Adding registry key for protocol handlers, and a custom action to remove registry keys on installation (including keys added by SDA, like auto-launch)
[SDA-2393](https://perzoinc.atlassian.net/browse/SDA-2393)

## Solution Approach
Describe the approach you've taken to implement this change / resolve the issue

## Related PRs
List related PRs against other branches/repositories:

branch | PR
------ | ------
other_pr_dev | [link]()

### CODE REVIEW

File: installer/win/WixSharpInstaller/Symphony.cs [hunk 1]
```diff
@@ -258,11 +273,68 @@ public static ActionResult UpdateConfig(Session session)
...
     }
 
+    // CleanRegistry custom action
+    [CustomAction]
+    public static ActionResult CleanRegistry(Session session)
+    {
+        try
+        {
+            // Remove registry keys added for auto-launch
+
```
Comment (VishwasShashidhar) [inline installer/win/WixSharpInstaller/Symphony.cs:290]: Is this blank line by mistake?

File: installer/win/WixSharpInstaller/Symphony.cs [hunk 2]
```diff
@@ -258,11 +273,68 @@ public static ActionResult UpdateConfig(Session session)
...
+            using( var key = Registry.CurrentUser.OpenSubKey(@"Software\Microsoft\Windows\CurrentVersion\Run", true) )
+            {
+                if (key != null)
+                {
+                    key.DeleteValue("Symphony", false);
```
Comment (VishwasShashidhar) [inline installer/win/WixSharpInstaller/Symphony.cs:295]: what is false in this case? Can we move it to a variable and use the variable here so the code is self explanatory?

Same for all the following occurrences.

File: installer/win/WixSharpInstaller/Symphony.cs [hunk 3]
```diff
@@ -258,11 +273,68 @@ public static ActionResult UpdateConfig(Session session)
...
+                if (key != null)
+                {
+                    key.DeleteSubKeyTree("symphony", false);
+                }
+            }
+        }
+        catch (System.Exception e)
+        {
+            session.Log("Error executing CleanRegistry: " + e.ToString() );
+            return ActionResult.Success;
```
Comment (VishwasShashidhar) [inline installer/win/WixSharpInstaller/Symphony.cs:336]: can we use finally and return the result in one step?

File: installer/win/WixSharpInstaller/Symphony.cs (hunk 1 above)
Comment (mattias-symphony) [reply installer/win/WixSharpInstaller/Symphony.cs:290]: No

File: installer/win/WixSharpInstaller/Symphony.cs (hunk 2 above)
Comment (mattias-symphony) [reply installer/win/WixSharpInstaller/Symphony.cs:295]: it is just the API for C# DeleteValue - second parameter is throwOnMissingKey or something like that. But I don't think we need a variable for this, I think it is a case where you would look up the API function anyway, as it is a Win32 API function and there are usually subtleties there if you want to understand what is happening

File: installer/win/WixSharpInstaller/Symphony.cs (hunk 3 above)
Comment (mattias-symphony) [reply installer/win/WixSharpInstaller/Symphony.cs:336]: yeah, but I think it is cleaner to do it like this, and more in line with the other custom actions which follows this pattern (as they can return ActionResult.Error)

Comment (VishwasShashidhar) [reply installer/win/WixSharpInstaller/Symphony.cs:336]: ok. It's a different way of coding anyway.

**Review by VishwasShashidhar — APPROVED** [review]

**Review by johankwarnmarksymphony — APPROVED** [review]

**Pull request merged**